store/src/main/scala/geotrellis/store/cog/COGLayerMetadata.scala

```diff
@@ -420,4 +400,25 @@ object COGLayerMetadata {
             throw DeserializationException(s"COGLayerMetadata expected, got $v")
         }
     }
+
+  implicit def cogLayerMetadataSemigroup[K: Boundable: SpatialComponent] = new Semigroup[COGLayerMetadata[K]] {
+    def combine(x: COGLayerMetadata[K], y: COGLayerMetadata[K]): COGLayerMetadata[K] = {
+      val combinedZoomRangeInfos =
+        (x.zoomRangeInfos ++ y.zoomRangeInfos)
+          .groupBy(_._1)
+          .map { case (key, bounds) => key -> bounds.map(_._2).reduce(_ combine _) }
+          .toVector
+          .sortBy(_._1)
+
+      val combinedExtent = x.extent.expandToInclude(y.extent)
```

Comment:
It can be `combine` in case we really need it.

Reply:
@pomadchin What do you mean? I don't follow.

Reply:
@echeipesh we have lots of semigroup instances for our types, but why do we need them if we don't use them? see: https://github.com/locationtech/geotrellis/pull/3002#discussion_r297680809